File: lib_common/src/main/java/com/wujia/lib_common/data/network/exception/ApiException.java
package com.wujia.lib_common.data.network.exception;


public class ApiException extends Exception {
    public String code;

    public ApiException(Throwable throwable) {
        super(throwable.getMessage(), throwable);
    }

    public ApiException(Throwable throwable, String code, String message) {
        super(message, throwable);
        this.code = code;
    }

    public boolean isNetworkConnectError() {
        return code == ERROR.NETWORK_CONNECT_ERROR ? true : false;
    }

}

File: lib_common/src/main/java/com/wujia/lib_common/data/network/exception/ApiException.kt
package com.wujia.lib_common.data.network.exception


class ApiException : Exception {
    var code: String?=null

    constructor(throwable: Throwable) : super(throwable.message, throwable)

    constructor(throwable: Throwable, code: String, message: String?) : super(message, throwable) {
        this.code = code
    }

}

